src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -29,8 +28,12 @@ public static class ResolverUtility
         {
             var key = new LibraryRangeCacheKey(libraryRange, framework);
 
-            return cache.GetOrAdd(key, (cacheKey) =>
-                FindLibraryEntryAsync(cacheKey.LibraryRange, framework, runtimeIdentifier, context, cancellationToken));
+            if (cache.TryGetValue(key, out var graphItem))
+                return graphItem;
+
+            _ = cache.TryAdd(key, FindLibraryEntryAsync(key.LibraryRange, framework, runtimeIdentifier, context, cancellationToken));
+
+            return cache[key];
```

Comment:
nit: You could've used get or add instead of try add.

Reply:
Fixed in the latest commit. I agree that using `GetOrAdd` is cleaner and I removed the usage of discard bool.